tests/test_quoting.py

```diff
@@ -455,3 +459,130 @@ def test_quoter_path_with_plus(quoter):
 def test_unquoter_path_with_plus(unquoter):
     s = "/test/x+y%2Bz/:+%2B/"
     assert "/test/x+y+z/:++/" == unquoter(unsafe="+")(s)
+
+
+@given(safe=st.text(), protected=st.text(), qs=st.booleans(), requote=st.booleans())
+def test_fuzz__PyQuoter(safe, protected, qs, requote):
+    _PyQuoter(safe=safe, protected=protected, qs=qs, requote=requote)
+
+
+@given(ignore=st.text(), unsafe=st.text(), qs=st.booleans())
+def test_fuzz__PyUnquoter(ignore, unsafe, qs):
+    _PyUnquoter(ignore=ignore, unsafe=unsafe, qs=qs)
+
+
+@example(text_input="0")
+@settings(suppress_health_check=(HealthCheck.function_scoped_fixture,))
+@given(
+    text_input=st.text(
+        alphabet=st.characters(max_codepoint=127, blacklist_characters="%")
+    )
+)
+def test_quote_unquote_parameter(
+    quoter: Type[_PyQuoter],
+    unquoter: Type[_PyUnquoter],
+    text_input: str,
+) -> None:
+    quote = quoter()
+    unquote = unquoter()
+    text_quoted = quote(text_input)
+    note(f"text_quoted={text_quoted!r}")
+    text_output = unquote(text_quoted)
+    assert text_input == text_output
+
+
+@example(text_input="0")
+@settings(suppress_health_check=(HealthCheck.function_scoped_fixture,))
+@given(
+    text_input=st.text(
+        alphabet=st.characters(max_codepoint=127, blacklist_characters="%")
+    )
+)
+def test_quote_unquote_parameter_requote(
+    quoter: Type[_PyQuoter],
+    unquoter: Type[_PyUnquoter],
+    text_input: str,
+) -> None:
+    quote = quoter(requote=True)
+    unquote = unquoter()
+    text_quoted = quote(text_input)
+    note(f"text_quoted={text_quoted!r}")
+    text_output = unquote(text_quoted)
+    assert text_input == text_output
+
+
+@example(text_input="0")
+@settings(suppress_health_check=(HealthCheck.function_scoped_fixture,))
+@given(
+    text_input=st.text(
+        alphabet=st.characters(max_codepoint=127, blacklist_characters="%")
+    )
+)
+def test_quote_unquote_parameter_path_safe(
+    quoter: Type[_PyQuoter],
+    unquoter: Type[_PyUnquoter],
+    text_input: str,
+) -> None:
+    quote = quoter()
+    unquote = unquoter(ignore="/%", unsafe="+")
+    assume("+" not in text_input and "/" not in text_input)
+    text_quoted = quote(text_input)
+    note(f"text_quoted={text_quoted!r}")
+    text_output = unquote(text_quoted)
+    assert text_input == text_output
+
+
+if not NO_EXTENSIONS and False:
```

Comment:
We should do
```suggestion
if not NO_EXTENSIONS:
```
at some point, that was a temporary hack.